backend/routes/crypto.js

```diff
@@ -0,0 +1,185 @@
+const randomstring = require('randomstring')
+
+const { authShop } = require('./_auth')
+const { Network, PaymentSession, Transaction } = require('../models')
+const { txQueue } = require('../queues/queues')
+const { decryptShopOfferData } = require('../utils/offer')
+
+const { getLogger } = require('../utils/logger')
+const { TransactionTypes, TransactionStatuses } = require('../enums')
+
+const log = getLogger('routes.crypto')
+
+// How long before a payment session is considered expired.
+const PAYMENT_SESSION_EXPIRY = 10 * 60 * 1000 // 10 min
+
+// Endpoints for handling direct crypto payments that do not go thru the
+// marketplace contract.
+// The flow is as follow:
+//   1. FE calls /crypto/payment-code to get a paymentCode
+//   2. FE sends the blockchain tx to the network and gets a tx hash back
+//   3. FE calls /crypto/payment with the tx hash and the paymentCode
+//   4. BE waits for the tx to get mined
+//   5. BE processes the order
+
+module.exports = function (router) {
+  /**
+   * Endpoint to request a payment code to the back-end.
+   *
+   * The body of the request is expected to include:
+   *   {string} fromAddress: Address sending the transaction.
+   *   {string} toAddress: Address receiving the transaction.
+   *   {string} amount: transaction amount
+   *   {string} currency: transaction currency (ex: ETH, OGN, etc...).
+   */
+  router.get('/crypto/payment-code', authShop, async (req, res) => {
```

Comment:
should be `.post`?

Reply:
yes good point since it writes data in the DB. let me update it...

Reply:
done